doc/rst/language/spec/modules.rst

```diff
@@ -327,38 +327,68 @@ access a module's symbols from outside of the module.  For top-level modules, a
-statement within a scope has no effect on its behavior.  If a scope includes
-multiple ``use`` statements, multiple ``import`` statements, or a combination of
-``import`` and ``use`` statements, then the newly-visible names are inserted
-into a common enclosing scope.
+The ``use`` and ``import`` statements themselves are processed in order,
+so it is not possible to ``use`` a module that is only made available by
+a later ``use``.
+
+The other mentiones of a name made visible by a ``use`` or ``import``
+statement can be at any position relative to the ``use`` or ``import``.
+
+Private ``use`` statements -- for example ``use M`` or ``private use M``
+- make the contents of the module available in a scope just outside of
+the current one and the name of the module itself (``M`` in the xample)
+available in a second scope just outside of that. In contrast, ``import``
+as well as ``public use`` do not use these implicit scopes.
 
 .. _Use_And_Import_Conflicts:
 
 Conflicts
 +++++++++
 
-The implicit scope added by ``use`` and ``import`` described in the previous
-section follows the same rules about conflicting variables as other scopes (see
+Variable names available through ``use`` or ``import``
+follow the same rules about conflicting variables as other scopes (see
 :ref:`Variable_Conflicts`).  Thus an error will be signaled if multiple
 variables with the same name would be inserted into this enclosing scope and
 that name is accessed.  Remember that this does not apply to functions unless
 they are also indistinguishable in other ways, see :ref:`Function_Overloading`.
 
-Because symbols brought into scope by a ``use`` or ``import`` statement are
+Because symbols brought into scope by a ``private use`` statement are
 placed at a scope enclosing where the statement appears, such symbols will be
 shadowed by other symbols with the same name defined in the scope with the
 statement.  The symbols that are shadowed will only be accessible via
-:ref:`Explicit_Naming`.
+:ref:`Explicit_Naming`. For example:
+
+   *Example (shadowing.chpl)*.
+
+   .. code-block:: chapel
+
+      module A {
+        var x: int;
+      }
 
-Symbols defined by public ``use`` or ``import`` statements can impact the scope
-they are inserted into in different ways (see :ref:`Public_Use` and
-:ref:`Reexporting` for more information on the ``public`` keyword).  Symbols
-that are brought in by a ``public use`` for unqualified access are treated as
-at successive distances relative to how many ``public use`` statements were
-necessary to obtain them.  For instance,
+      module MainMod {
+        use A;
+        var x = "hello";
+
+        proc main() {
+          writeln(x);
+        }
+      }
+
+   This program will compile and print out ``hello`` because the use of
+   ``x`` refers to ``MainMod.x`` which shadows ``A.x`` because ``use A``,
+   which means the same as ``private use A``, introduces ``x`` in a scope
```

Comment:
This is a bit of a run on sentence.  Maybe just make the `use` private in the example?  I'm sure we talk about the default privacy somewhere else so it's okay to not get into it.